src/middleware/fronteggMiddleware.ts

```diff
@@ -56,20 +73,33 @@ export function frontegg(options: IFronteggOptions) {
     throw new Error('Missing context resolver');
   }
 
-
   const authenticator = new FronteggAuthenticator();
 
   ContextHolder.setContext({
     FRONTEGG_CLIENT_ID: options.clientId,
     FRONTEGG_API_KEY: options.apiKey,
   });
 
-  let authInitedPromise = authenticator.init(options.clientId, options.apiKey);
+  let authInitedPromise = authenticator.init(
+    options.clientId,
+    options.apiKey
+  );
 
   proxy.on('error', async (err, req: any, res, _) => {
     Logger.error(`Failed proxy request to ${req.url} - `, err);
     req.frontegg.retryCount++;
-    Logger.info(`retry count of ${req.url} - `, req.frontegg.retryCount);
+    Logger.info(
+      `retry count of ${req.url} - `,
+      req.frontegg.retryCount
+    );
+
+    if (res.headersSent) {
```

Comment:
Following our chat, please explain how it was tested and check if we should still run the code although headers were sent.
We might do that by replacing the function 'writeHead' in line #105 with 'statusCode'